packages/office-ui-fabric-react/src/components/SwatchColorPicker/SwatchColorPicker.tsx

```diff
@@ -0,0 +1,580 @@
+
+  /**
+   * Get the selected item's index
+   * @param items - The items to search
+   * @param selectedId - The selected item's id to find
+   * @returns {number} - The index of the selected item's id, -1 if there was no match
+   */
+  private _getSelectedIndex(items: ISwatchColorPickerItemProps[], selectedId: string): number {
+    return findIndex(items, (item => (item.id === selectedId)));
+  }
+
+  /**
+   * Render the swatch color picker with a button in the given (collapased vs expanded state)
+   * @returns {JSX.Element[]} - an array of the elements to render
+   */
+  private _buttonToRender(): JSX.Element[] {
+    let { selectedIndex, expanded } = this.state;
+    let {
+      swatchColorPickerButtonProps,
+      swatchColorPickerItems,
+      updateButtonIconWithColor,
+      width,
+      disabled
+    } = this.props;
+
+    let colorToSet = null;
+
+    // Do we need to update the button with the selected
+    // item's color? If so, attempt to grab the color
+    if (updateButtonIconWithColor &&
+      selectedIndex &&
+      selectedIndex >= 0 &&
+      selectedIndex < swatchColorPickerItems.length) {
+      let swatchColorPickerItem = swatchColorPickerItems[selectedIndex];
+      if (swatchColorPickerItem.type === SwatchColorPickerItemType.Cell &&
+        swatchColorPickerItem.color !== null &&
+        swatchColorPickerItem.color.length > 0) {
+        colorToSet = swatchColorPickerItem.color;
+      }
+    }
+
+    let elements: JSX.Element[] = [];
+
+    elements.push(<DefaultButton
+      { ...swatchColorPickerButtonProps }
+      style={ { color: colorToSet && colorToSet } }
+      className={
+        css('ms-swatchColorPickerButton',
+          {
+            'is-expanded': expanded
+          }
+        )
+      }
+      onClick={ this._onButtonClick }
+      aria-haspopup={ true }
+      aria-expanded={ (!this.props.disabled && this.state.expanded) ? true : false }
+      disabled={ this.props.disabled }
+      menuIconProps={ !swatchColorPickerButtonProps.menuIconProps ? { iconName: 'chevronDown' } : swatchColorPickerButtonProps.menuIconProps }
+    />);
+
+    // Add the menu if we are endabled and expanded
+    if (!this.props.disabled && this.state.expanded) {
+      elements.push(this._onRenderContainer());
+    }
+
+    return elements;
+  }
+
+  /**
+   * Render the full swatch color picker this happens for both
+   * the always visible case as well as the expandable case (when rendered in a menu)
+   * @returns {JSX.Element} - The element representing the fully rendered swatch color picker
+   */
+  @autobind
+  private _fullSwatchColorPickerToRender(): JSX.Element {
+    let { swatchColorPickerItems, width } = this.props;
+    return (
+      <FocusZone
+        isCircularNavigation={ true }
+        className={ css('ms-swatchColorPickerBodyContainer', styles.swatchColorPickerContainer) }
+      >
+        { this._onRenderItems(swatchColorPickerItems.map((item, index) => { return { ...item, index }; })) }
```

Comment:
Yes, it is effectively overriding the index if it was set, but there's no expectation of index does outside of the component. This ensures that we have good indices on the items when processing farther down. Note, this is specifically important when mapping the selected index.

If we didn't have it on the props, we'd have to pass it down which kind of defeats the benefits of using map here